Added recommendations for code formatting and styling

owaiskazi19 opened the pull request:
Signed-off-by: Owais Kazi <owaiskazi19@gmail.com>

### Description
Added step wise guidelines and recommendation for code styling plugins.
 
### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/1362
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

saratvemulapalli commented on STANDARDS.md:
```diff
@@ -32,13 +32,34 @@ Make effective use of namespacing since it will allow co-existence of ambiguous
 
+How plugins were using CheckStyle?
+
+Plugins were using CheckStyle formatting plugin present in OpenSearch Core’s build.gradle. When core decided to go with Spotless and removed CheckStyle plugin, it caused issue on plugins side and the build broke as CheckStyle plugin was not available to format the code on plugin side. 
```
Time in context does not help.
Could you list down versions of OpenSearch where checkstyle is deprecated ?
And also link to the issue, it would help plugin developers understand why opensearch made the change.

owaiskazi19 replied:
Added the version of OpenSearch in which the changes were made and linked the issue.

saratvemulapalli reviewed (dismissed):
Thanks @owaiskazi19 for this!

saratvemulapalli commented:
@dblock / @CEHENKLE can we get another review on this?

dblock reviewed (changes requested):
I think we should be more prescriptive. We don't want half the plugins to be using Checkstyle and half the plugins to be using Spotless. I would say "code styling is done with spotless, and here's how", then remove everything related to Checkstyle since we don't want anyone to use, re move all the explanations of why we chose this vs. that, move the explanation into issue #1362 and link it from the doc with "for more information see #1362". WDYT?

owaiskazi19 commented:
> I think we should be more prescriptive. We don't want half the plugins to be using Checkstyle and half the plugins to be using Spotless. I would say "code styling is done with spotless, and here's how", then remove everything related to Checkstyle since we don't want anyone to use, re move all the explanations of why we chose this vs. that, move the explanation into issue #1362 and link it from the doc with "for more information see #1362". WDYT?

Makes sense. I have added the details in https://github.com/opensearch-project/OpenSearch/issues/1362 and updated the PR.
